src/tools/clippy/clippy_lints/src/lib.rs

```diff
@@ -17,6 +18,8 @@
 #![warn(rust_2018_idioms, unused_lifetimes)]
 // warn on rustc internal lints
 #![warn(rustc::internal)]
+// Disable this rustc lint for now, as it was also done in rustc
+#![allow(rustc::potential_query_instability)]
```

Comment:
ping @lcnr regarding #92584: I also allowed this lint in Clippy for now, since I don't know what to really do about this.

Reply:
ah :thinking: clippy shouldn't care about it afaict, that's only an issue inside of rustcs query system

once the lint is part of the beta compiler i am going to actually start using it in rustc itself

Reply:
Yeah, that's also how I understood it, but wasn't 100% sure. I'll change this comment in the next sync. :+1: